### ISSUE TITLE
[Patterns] Migrate old pattern pages to new style and format

### ISSUE THREAD

**Issue by johanstromqvist** [Shopify/polaris-react-archive#10906]
**TL;DR** The Patterns section is a mess making it hard for builders to understand and use. The pages should be updated to make sure builders can use them to get their job done, and the pages should use the new style and format that was developed for Uplift Polaris Documentation.

| Section | Screenshot |
|--------|--------|
| [New style](https://polaris.shopify.com/design/colors) | ![Image](https://github.com/Shopify/polaris/assets/38329124/ed752634-47be-41d2-8a33-76f9511e3d81) |
| [Old Patterns](https://polaris.shopify.com/patterns/new-features) | ![Image](https://github.com/Shopify/polaris/assets/38329124/04c8b1ad-f8ba-497f-a5ca-76fa47e33cac) |
| [New Patterns](https://polaris.shopify.com/patterns/date-picking/single-date) | ![Image](https://github.com/Shopify/polaris/assets/38329124/c135672e-b3b4-455f-9b7c-832cee1a0992) | 

### Success criteria
- Similar consumer experience as [the new docs](https://polaris.shopify.com/design/pro-design-language)
  - Tabbed page hierarchy
  - Same page layout and hierarchy 
  - 40:60 text-to-image ratio
- Meet similar requirements as the [current patterns](https://polaris.shopify.com/patterns/app-settings-layout)
  - **Choose:** It's easy for builders to understand if the pattern solves the merchant problem at hand
  - **Build:** It's easy for builders to find the right code resources to build the pattern quickly
  - **Configure:** It's easy for builders to understand what critical decisions they need to make to build the pattern accurately
